### PR TITLE
Add relative positional embedding to KerasBERT

### PR DESCRIPTION
**Pull request by xinliupitt**
# Description

> :memo: I added the RelativePositionEmbedding class to layers/position_embedding.py. This class generates sine cosine relative positional embedding. Such embedding is defined and formulized in "Attention is All You Need", section 3.5.
>  
> * In this way, nlp/transformer can directly use RelativePositionEmbedding under KerasBERT library.

## Type of change

- [x] TensorFlow 2 migration

## Tests

> :memo: Please describe the tests that you ran to verify your changes.
>  
> * I added two test functions in layers/position_embedding_test.py. Tests were successful.
> * nlp/transformer_main_test.py was changed to call RelativePositionEmbedding(). This test file was run without errors.

**Test Configuration**:

## Checklist

- [ ] I have signed the [Contributor License Agreement](https://github.com/tensorflow/models/wiki/Contributor-License-Agreements).
- [ ] I have read [guidelines for pull request](https://github.com/tensorflow/models/wiki/Submitting-a-pull-request).
- [ ] My code follows the [coding guidelines](https://github.com/tensorflow/models/wiki/Coding-guidelines).
- [ ] I have performed a self [code review](https://github.com/tensorflow/models/wiki/Code-review) of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by saberkun** [comment]
Some lint errors:
FAIL: Found 16 non-whitelited pylint errors:
official/nlp/modeling/layers/position_embedding.py:127: [C0301(line-too-long), ] Line too long (83/80)

official/nlp/modeling/layers/position_embedding.py:128: [C0301(line-too-long), ] Line too long (84/80)

official/nlp/modeling/layers/position_embedding.py:146: [C0301(line-too-long), ] Line too long (82/80)

official/nlp/modeling/layers/position_embedding.py:176: [C0301(line-too-long), ] Line too long (83/80)

official/nlp/modeling/layers/position_embedding.py:188: [C0301(line-too-long), ] Line too long (81/80)

official/nlp/modeling/layers/position_embedding.py:189: [C0301(line-too-long), ] Line too long (81/80)

official/nlp/modeling/layers/position_embedding.py:190: [C0301(line-too-long), ] Line too long (87/80)

official/nlp/modeling/layers/position_embedding.py:164: [E1003(bad-super-call), RelativePositionEmbedding.get_config] Bad first argument 'PositionEmbedding' given to super()

official/nlp/modeling/layers/position_embedding_test.py:104: [C0330(bad-continuation), ] Wrong hanging indentation (add 4 spaces).

official/nlp/modeling/layers/position_embedding_test.py:121: [C0330(bad-continuation), ] Wrong hanging indentation (add 4 spaces).

File: official/nlp/transformer/transformer.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
 from official.nlp.transformer import metrics
 from official.nlp.transformer import model_utils
 from official.nlp.transformer.utils.tokenizer import EOS_ID
-
+from official.nlp.modeling.layers import position_embedding
```
Comment (saberkun) [inline official/nlp/transformer/transformer.py:32]: the import should follow the order.

File: official/nlp/modeling/layers/position_embedding.py [hunk 2]
```diff
@@ -118,3 +120,75 @@ def call(self, inputs):
       position_embeddings = self._position_embeddings
 
     return tf.broadcast_to(position_embeddings, input_shape)
+
+@tf.keras.utils.register_keras_serializable(package="Text")
+class RelativePositionEmbedding(tf.keras.layers.Layer):
+  """Creates a positional embedding.
+  This layer calculates the position encoding as a mix of sine and cosine
```
Comment (chenGitHuber) [inline official/nlp/modeling/layers/position_embedding.py:127]: sine -> sin

File: official/nlp/modeling/layers/position_embedding.py [hunk 3]
```diff
@@ -118,3 +120,75 @@ def call(self, inputs):
...
+    hidden_size: Size of the hidden layer.
+    min_timescale: Minimum scale that will be applied at each position
+    max_timescale: Maximum scale that will be applied at each position.
+  """
+
+  def __init__(self,
+               hidden_size,
+               min_timescale=1.0,
+               max_timescale=1.0e4,
+               length=None,
```
Comment (chenGitHuber) [inline official/nlp/modeling/layers/position_embedding.py:141]: please explain the "length" argument in docstring.

File: official/nlp/modeling/layers/position_embedding.py [hunk 4]
```diff
@@ -118,3 +120,75 @@ def call(self, inputs):
...
+  def call(self, inputs):
+    """Implements call() for the layer."""
+    length = self._length
+    if inputs is None and length is None:
+      raise ValueError(
+          "If inputs is None, `length` must be set in "
+          "RelativePositionEmbedding().")
+    if inputs is not None:
+      input_shape = tf_utils.get_shape_list(inputs)
+      length = input_shape[1]
```
Comment (chenGitHuber) [inline official/nlp/modeling/layers/position_embedding.py:180]: how about raising an error if self._length is also specified, but not equal to input_shape[1]?

**Pull request merged**